gluster/fs/pkg/allocator/minmax.go

```diff
@@ -0,0 +1,191 @@
+/*
```

Comment:
is this the same as aws/efs allocator or copied from upstream allocator? (either way it's fine, I will be putting the allocator in `lib` soon)